devito/types/relational.py

```diff
@@ -0,0 +1,189 @@
+"""User API to specify relationals."""
+
+import sympy
+
+
+__all__ = ['Le', 'Lt', 'Ge', 'Gt']
+
+
+class Le(sympy.Le):
+    """
+    A less-than ("<=") relation between two objects, the left-hand side and the
+    right-hand side. Can be used to build conditionals but not directly to
+    construct an Operator.
+
+    The left-hand side may be a Function or a SparseFunction. The right-hand
+    side may be any arbitrary expressions with numbers, Dimensions, Constants,
+    Functions and SparseFunctions as operands.
+
+    Parameters
+    ----------
+    lhs : Function or SparseFunction
```

Comment:
Can't be an `Expr` right?

Reply:
it should. Try, perhaps add test, and change the docstring as "expr-like"

Reply:
Added a test and seems to be all fine.

Reply:
Not sure yet if it works for subdomain correctly. Getting some random(OMG!) failures!